### DISCUSSION TITLE
License Key - Devices

### DISCUSSION THREAD

**Discussion by SlugCamargo** [Able-Bear-Software/tab-nodes-tree#26] [Bug Report]
I'm posting this as a bug discussion because I'm not sure where it would really go. It's more of a support issue, I guess.

SO! I have a work computer and a personal one. In each one of them I switch between my work and personal Chrome profiles. 

I bought two licenses, I figured it would go "the personal computer license" and "the work computer license"; but each one is being counted as used per device ***and*** per profile - So I can't use my work license for both profiles, even though they are in the same computer, in the same Windows login, just different Chrome profiles. 

Is that how it should work? Should I buy 4 licenses in order to use the same 2 profiles across 2 devices?

**Comment by zeroarst** [comment]
Hi @SlugCamargo,

You’re right—the license is currently tied to each Chrome profile on a per-device basis. Let me explain why it works this way.

To authorize a device, the license needs to be associated with a unique ID to ensure it's linked to the correct entity. Initially, I intended to bind the license per device so that users could use the same license key across multiple Chrome profiles on the same machine. However, due to technical limitations, there’s no way for developers to retrieve a unique device ID through the browser or Chrome APIs.

As a result, I had to generate this ID myself and store it per profile, since each Chrome profile has its own storage. To the best of my knowledge, there’s no way to access the same storage across different profiles.

That said, I have an idea: it might be possible to sync the unique ID across different devices using the same user profile. However, this would require significant development and testing, and I can't guarantee it will work or be implemented soon.

I hope this clarifies the situation, and thank you for your understanding!

**Reply by SlugCamargo** [reply]
I see. Well, it's not that big a deal; the license is pretty cheap anyway, especially considering how much the extension improved my workflow already.
